pennylane/interfaces/batch/__init__.py

```diff
@@ -101,11 +232,15 @@ def cost_fn(params, x):
            [ 0.01983384, -0.97517033,  0.        ],
            [ 0.        ,  0.        , -0.95533649]])
     """
-    # Default execution function; simply call device.batch_execute
-    # and return no Jacobians.
-    execute_fn = lambda tapes, **kwargs: (device.batch_execute(tapes), [])
     gradient_kwargs = gradient_kwargs or {}
 
+    if isinstance(cache, bool) and cache:
+        # cache=True: create a LRUCache object
+        cache = LRUCache(maxsize=cachesize, getsizeof=len)

```

Comment:
Oh, so this is the reason for the `Cache` type in the docstrings!

Reply:
kind of, but it is very much leaning into ducktyping!

e.g., `dict()` is also a valid cache. In fact, anything that defines `pop`, `get` and `setdefault` is a cache